packages/api/src/router/user.ts

```diff
@@ -36,6 +36,94 @@ export const userRouter = router({
     return user;
   }),
 
+  validateTrialOrSub: privateProcedure.query(async ({ ctx }) => {
```

Comment:
This sounds like we have our own `AgreetoSubscription` model now with a custom `AgreetoSubscriptionStatus` enum similar to how @bgavas had from the beginning. 

Another idea for the name that came to my mind is `agreetoSubscription`.